### ISSUE TITLE
Implement `safe_softmax `

### ISSUE THREAD

**Issue by veritas9872** [huggingface/optimum-habana#1843]
### Feature request

The README mentions that a `safe_softmax` function will be implemented but there is no mention of this function in v1.16.0. What is the timeline for the implementation?

### Motivation

The decode phase of LLM inference is bottlenecked by softmax. An efficient implementation is urgent.

### Your contribution

Maybe help time the function?

**Comment by libinta** [comment]
@veritas9872 thanks for the query. The operator _safe_softmax implementation is in synapse 1.20. you can check [_safe_softmax](https://docs.habana.ai/en/latest/PyTorch/Reference/Pytorch_Operators/Pytorch_Operators.html?highlight=pytorch%20operators%20support%20summary#pytorch-operators-support-summary) to find details.

**Comment by libinta** [comment]
@veritas9872  in OH, we normally use the fusedsdpa in attention for performance and memory benefit. If you choose to use sdpa (transformers eager attn implementation), then pytorch scaled_dot_product_attention is used, and pytorch scaled_dot_product_attention uses safe_softmax op, so it's transparent to user. No need for any change from user perspective.  In synapse 1.20, we optimized the safe_softmax kernel for performance.
Thanks.
